test/Libraries/PackageManagerTests/PackageLoaderTests.cs

```diff
@@ -110,13 +110,31 @@ public void ScanPackageDirectoryReturnsNullForInvalidDirectory()
         public void LoadPackagesReturnsAllValidPackagesInValidDirectory()
         {
             var loader = new PackageLoader(PackagesDirectory);
+            var libraryLoader = new ExtensionLibraryLoader(CurrentDynamoModel);
+
+            loader.PackagesLoaded += libraryLoader.LoadPackages;
+            loader.RequestLoadNodeLibrary += libraryLoader.LoadNodeLibrary;
+
             loader.LoadAll(new LoadPackageParams
             {
-                Preferences = this.CurrentDynamoModel.PreferenceSettings
+                Preferences = CurrentDynamoModel.PreferenceSettings,
+                PathManager = CurrentDynamoModel.PathManager
             });
 
-            // There are 8 packages in "GitHub\Dynamo\test\pkgs"
-            Assert.AreEqual(8, loader.LocalPackages.Count());
+            // There are 11 packages in "GitHub\Dynamo\test\pkgs"
+            Assert.AreEqual(11, loader.LocalPackages.Count());
+
+            // Verify that interdependent packages are resolved successfully
+            var libs = CurrentDynamoModel.LibraryServices.ImportedLibraries.ToList();
+            Assert.IsTrue(libs.Any(x => File.Exists(Path.Combine(PackagesDirectory, "AnotherPackage", "bin", "AnotherPackage.dll"))));
+            Assert.IsTrue(libs.Any(x => File.Exists(Path.Combine(PackagesDirectory, "Dependent Package", "bin", "DependentPackage.dll"))));
+            Assert.IsTrue(libs.Any(x => File.Exists(Path.Combine(PackagesDirectory, "Package", "bin", "Package.dll"))));
+
+            // Verify that interdependent packages are imported successfully
+            var entries = CurrentDynamoModel.SearchModel.SearchEntries.ToList();
+            Assert.IsTrue(entries.Any(x => x.FullName == "AnotherPackage.AnotherPackage.AnotherPackage.HelloAnotherWorld"));
+            Assert.IsTrue(entries.Any(x => x.FullName == "DependentPackage.DependentPackage.DependentPackage.HelloWorld"));
+            Assert.IsTrue(entries.Any(x => x.FullName == "Package.Package.Package.Hello"));
```

Comment:
The following code shows the methods defined in the 3 test packages. `DependentPackage` has a dependency on `Package` and `AnotherPackage` has a dependency on the first two:
![image](https://user-images.githubusercontent.com/5710686/59473203-62844500-8e0f-11e9-9149-320ed98bd44f.png)